examples/android/clientcache/app/src/test/java/io/grpc/clientcacheexample/SafeMethodCachingInterceptorTest.java

```diff
@@ -0,0 +1,533 @@
+    try {
+      ClientCalls.blockingUnaryCall(
+          channelToUse,
+          safeGreeterSayHelloMethod,
+          CallOptions.DEFAULT.withOption(
+              SafeMethodCachingInterceptor.ONLY_IF_CACHED_CALL_OPTION, true),
+          message);
+      fail("Expected call to fail");
+    } catch (StatusRuntimeException sre) {
+      assertEquals(Status.UNAVAILABLE.getCode(), sre.getStatus().getCode());
+      assertEquals(
+          "Unsatisfiable Request (only-if-cached set, but value not in cache)",
+          sre.getStatus().getDescription());
+    }
+  }
+
+  @Test
+  public void requestWithOnlyIfCachedOption_usesCache() {
+    HelloReply reply1 =
+        ClientCalls.blockingUnaryCall(
+            channelToUse, safeGreeterSayHelloMethod, CallOptions.DEFAULT, message);
+    HelloReply reply2 =
+        ClientCalls.blockingUnaryCall(
+            channelToUse,
+            safeGreeterSayHelloMethod,
+            CallOptions.DEFAULT.withOption(
+                SafeMethodCachingInterceptor.ONLY_IF_CACHED_CALL_OPTION, true),
+            message);
+
+    assertEquals(reply1, reply2);
+  }
+
+  @Test
+  public void requestWithNoCacheAndOnlyIfCached_fails() {
+    try {
+      ClientCalls.blockingUnaryCall(
+          channelToUse,
+          safeGreeterSayHelloMethod,
+          CallOptions.DEFAULT
+              .withOption(SafeMethodCachingInterceptor.NO_CACHE_CALL_OPTION, true)
+              .withOption(SafeMethodCachingInterceptor.ONLY_IF_CACHED_CALL_OPTION, true),
+          message);
+      fail("Expected call to fail");
+    } catch (StatusRuntimeException sre) {
+      assertEquals(Status.UNAVAILABLE.getCode(), sre.getStatus().getCode());
+      assertEquals(
+          "Unsatisfiable Request (no-cache and only-if-cached conflict)",
+          sre.getStatus().getDescription());
+    }
+  }
+
+  @Test
+  public void responseNoCacheDirective_notCached() throws Exception {
+    cacheControlDirectives.add("no-cache");
+
+    HelloReply reply1 =
+        ClientCalls.blockingUnaryCall(
+            channelToUse, safeGreeterSayHelloMethod, CallOptions.DEFAULT, message);
+    HelloReply reply2 =
+        ClientCalls.blockingUnaryCall(
+            channelToUse, safeGreeterSayHelloMethod, CallOptions.DEFAULT, message);
+
+    assertNotEquals(reply1, reply2);
+    assertNotEquals(reply1, reply2);
+    assertEquals(0, cache.internalCache.size());
+    assertEquals(0, cache.removedKeys.size());
+  }
+
+  @Test
+  public void responseNoStoreDirective_notCached() throws Exception {
+    cacheControlDirectives.add("no-store");
+
+    HelloReply reply1 =
+        ClientCalls.blockingUnaryCall(
+            channelToUse, safeGreeterSayHelloMethod, CallOptions.DEFAULT, message);
+    HelloReply reply2 =
+        ClientCalls.blockingUnaryCall(
+            channelToUse, safeGreeterSayHelloMethod, CallOptions.DEFAULT, message);
+
+    assertNotEquals(reply1, reply2);
+    assertEquals(0, cache.internalCache.size());
```

Comment:
use Truth.assertThat(blah).isEmpty()

This will include what was present in the error message.

Reply:
Done